### ISSUE TITLE
public setting in CMS_LANGUAGES is ignored

### ISSUE THREAD

**Issue by zanderle** [divio/django-cms#4176]
I set `public` to `False` on two languages, yet they are displayed everywhere just the same as before. They are displayed in the toolbar as well as the language chooser.

Am I missing something or is this a bug? Might also be an unclear documentation...

Below are the settings used.
I am using Django 1.6 (I know, I know) and Django-CMS 3.0.12

```python
LANGUAGES = (
    ('en', gettext('English')),
    ('de', gettext('German')),
    ('sl', gettext('Slovenian')),
    ('hr', gettext('Croatian')),
)

CMS_LANGUAGES = {
    'default': {
        'fallbacks': ['en', 'sl'],
        'public': True,
        'hide_untranslated': False,
        'redirect_on_fallback': True,
    },
    1: [
        {
            'public': True,
            'code': 'en',
            'hide_untranslated': False,
            'name': gettext('English'),
            'redirect_on_fallback': True,
        },
        {
            'public': False,
            'code': 'de',
            'hide_untranslated': False,
            'name': gettext('German'),
            'redirect_on_fallback': True,
        },
        {
            'public': False,
            'code': 'sl',
            'hide_untranslated': False,
            'name': gettext('Slovenian'),
            'redirect_on_fallback': True,
        },
        {
            'public': True,
            'code': 'hr',
            'hide_untranslated': False,
            'name': gettext('Croatian'),
            'redirect_on_fallback': True,
        },
    ],
}
```

**Comment by yakky** [comment]
Public hides languages for non-staff users, while still making them accessible for editors to create the different languages, so it's a normal behavior to have all the languages available in the toolbar as it's part of the editor interface and they are supposed to access the 'hidden' languages to populate them

**Comment by zanderle** [comment]
@yakky ok that makes sense. But why is it shown in the language chooser? I would expect it to be hidden in the language chooser.

**Comment by mkoistinen** [comment]
Hi @zanderle, I've investigated this.

Can you create a new template in your project at templates/menu/language_chooser.html with these contents and let me know if it helps:

````
{% load i18n menu_tags %}

{% if languages|length > 1 %}
{% for language in languages %}
<li class="lang{% ifequal current_language language.0 %} active{% endifequal %}">
	<a href="{% page_language_url language.0 %}" title="{% trans "Change to language:" %} {{ language.1 }}">{{ language.0 }}</a>
</li>
{% endfor %}
{% endif %}
````

Please advise.

**Comment by zanderle** [comment]
@mkoistinen YES! Thank you so much! As soon as I remove `{% get_available_languages as languages %}` it works as expected - it shows only the translated languages vs. all of them.

This should probably be removed from the default template as well? Or at least documented somewhere...

**Comment by yakky** [comment]
Oh well, I thought the language chooser template has been fixed **ages** ago!

**Comment by yakky** [comment]
I'll fix it

**Comment by mkoistinen** [comment]
Crazy thing is, the templatetag _was_ fixed ages ago. But, for some bizarre reason, someone overrode all that work with one line in the template itself. Go figure. Amazingly, the person who did the work, was also the person who then overrode it. Weirder still.

**Comment by yakky** [comment]
Fixed by #4217
